### DISCUSSION TITLE
Is there a way to get the AST?

### DISCUSSION THREAD

**Discussion by Kreijstal** [markedjs/marked#2584] [Q&A]
I am not interested in a markdown to html converter, I am interested in a markdown parser, for scrapping purposes.

**Comment by UziTech** [comment]
You can get the tokens from the [lexer](https://marked.js.org/using_pro#lexer):

```js
const tokens = marked.lexer(md, options);
```

**Reply by Kreijstal** [reply]
I guess that's just tokenization... but thanks

**Reply by calculuschild** [reply]
Yep, the tokens object should be close to the parsing tree you are looking for.
